MapboxNavigation/NavigationView.swift

```diff
@@ -127,17 +155,27 @@ open class NavigationView: UIView {
     }
     
     func commonInit() {
+        floatingButtons = ["overviewButton": overviewButton, "muteButton": muteButton, "reportButton": reportButton]
         setupViews()
         setupConstraints()
     }
     
+    func clearStackViews( _ floatingButtons: [String: Button]?) {
+        let oldFloatingButtons: [UIView] = floatingStackView.subviews
+        for floatingButton in oldFloatingButtons {
+            floatingStackView.removeArrangedSubview(floatingButton)
+            floatingButton.removeFromSuperview()
+        }
+    }
+    
     func setupStackViews() {
-        floatingStackView.addArrangedSubviews([overviewButton, muteButton, reportButton])
+        if let buttons = floatingButtons {
+            let newFloatingButtons = Array(buttons.values)
```

Comment:
`Dictionary.values` isn’t guaranteed to be in any particular order. In the screenshot above, for example, the mute button appears below the feedback button, opposite of the default order on main. Consider using an array to store the buttons instead of a dictionary. If you need to associate each button with an ID of some sort, there are multiple possible approaches, such as making `floatingButtons` a dictionary of tuples or making the IDs properties of the respective button classes.

Reply:
The sequence of the dictionary could be fixed by replacing the `Array(Dictionary.values)` with `Dictionary.map{$0.value}`. But the problem of this design has some other problem related with the key value assigning as abobe. I would go back to the button arrays instead of the hashMap. Thanks for the suggestion!